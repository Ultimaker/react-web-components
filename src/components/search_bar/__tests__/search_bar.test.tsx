import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { SearchBar } from '..';

jest.mock('uuid/v4', () => jest.fn().mockReturnValue('d2e08c2a-365b-40a5-9f2d-64c28610b113'));

// Use real timers for this one because https://github.com/facebook/jest/issues/3465

describe('search_bar.tsx', () => {
    let wrapper: ReactWrapper;
    let emitDelay: number;
    const query = 'hello world';
    const placeholder = 'For exampleâ€¦';
    const onChange: jest.Mock<any, any> = jest.fn();

    afterEach(() => {
        onChange.mockReset();
        wrapper.unmount();
    });

    test('snapshot and instance', () => {
        wrapper = mount(<SearchBar onChange={onChange} placeholder={placeholder} />);
        expect(wrapper.instance()).toBeInstanceOf(SearchBar);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('label').prop('htmlFor')).toBe('d2e08c2a-365b-40a5-9f2d-64c28610b113');
        expect(wrapper.find('input').prop('id')).toBe('d2e08c2a-365b-40a5-9f2d-64c28610b113');
    });

    test('with "query" prop', (done) => {
        wrapper = mount(
            <SearchBar
                onChange={onChange}
                placeholder={placeholder}
                query={query}
            />,
        );
        expect(onChange).not.toHaveBeenCalled();
        setTimeout(() => {
            expect(onChange).toHaveBeenCalledTimes(1);
            expect(onChange).toHaveBeenCalledWith(query);
            done();
        }, 0);
    });

    test('with "emitDelay" prop', (done) => {
        emitDelay = 500;
        wrapper = mount(
            <SearchBar
                onChange={onChange}
                placeholder={placeholder}
                emitDelay={emitDelay}
                query={query}
            />,
        );
        expect(onChange).not.toHaveBeenCalled();
        setTimeout(() => {
            expect(onChange).toHaveBeenCalledTimes(1);
            expect(onChange).toHaveBeenCalledWith(query);
            done();
        }, emitDelay);
    });
});
