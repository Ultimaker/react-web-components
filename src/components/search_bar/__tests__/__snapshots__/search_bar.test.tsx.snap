// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search_bar.tsx snapshot and instance 1`] = `
<SearchBar
  emitDelay={0}
  onChange={[MockFunction]}
  placeholder="For example…"
>
  <div
    className="search-bar"
  >
    <span
      className="search-bar__icon"
    >
      <MagnifierSmall>
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15,1.99999996 C11.1,1.99999996 7.99999996,5.09999997 7.99999996,8.99999996 C7.99999996,10.6 8.49999996,12 9.39999999,13.2 L1.99999996,20.6 L3.39999996,22 L10.8,14.6 C12,15.5 13.4,16 15,16 C18.9,16 22,12.9 22,8.99999996 C22,5.09999997 18.9,1.99999996 15,1.99999996 Z M15,14 C12.2,14 9.99999999,11.8 9.99999999,8.99999997 C9.99999999,6.19999998 12.2,3.99999997 15,3.99999997 C17.8,3.99999997 20,6.19999998 20,8.99999997 C20,11.8 17.8,14 15,14 Z"
            fill="currentColor"
          />
        </svg>
      </MagnifierSmall>
    </span>
    <label
      className="search-bar__label"
      htmlFor="d2e08c2a-365b-40a5-9f2d-64c28610b113"
    >
      For example…
    </label>
    <ForwardRef
      className="search-bar__input"
      hasError={false}
      id="d2e08c2a-365b-40a5-9f2d-64c28610b113"
      onChange={[Function]}
      placeholder="For example…"
      type="text"
      value=""
    >
      <input
        className="input-text search-bar__input"
        id="d2e08c2a-365b-40a5-9f2d-64c28610b113"
        onChange={[Function]}
        placeholder="For example…"
        type="text"
        value=""
      />
    </ForwardRef>
  </div>
</SearchBar>
`;
